terraform/modules/private-dns-zone-extend/main.tf

```diff
@@ -0,0 +1,34 @@
+provider "aws" {
+  alias = "core-network-services"
+  version = "~> 2.0"
+}
+
+resource "aws_route53_vpc_association_authorization" "private_zone_vpc_auth" {
+provider = aws.core-network-services
+vpc_id   = var.vpc_id
+zone_id  = var.business_unit_name
```

Comment:
As below I'd pass the zone ID in from the output of the data call commented out below.

Reply:
this has now been corrected to be passed from the data entry within the module